src/main/kotlin/lotto/LottoNumber.kt

```diff
@@ -0,0 +1,14 @@
+package lotto
+
+class LottoNumber private constructor(val value: Int) {
+    companion object {
+        const val MINIMUM_NUMBER = 1
+        const val MAXIMUM_NUMBER = 45
+        private val NUMBERS: Map<Int, LottoNumber> =
+            (MINIMUM_NUMBER..MAXIMUM_NUMBER).associateWith(::LottoNumber)
+
+        fun from(value: Int): LottoNumber {
+            return NUMBERS[value] ?: throw IllegalArgumentException()
+        }
+    }
```

Comment:
VO Caching 👍 👍 💯 
so now if `LottoNumber` data type change from `Int` to `Long`, just change `LottoNumber` class only! 😄

Reply:
What are the differences between `class`, `value class`, `data class`?